rewrite-maven/src/main/java/org/openrewrite/maven/utilities/MavenArtifactDownloader.java

```diff
@@ -108,7 +108,7 @@ public Path downloadArtifact(ResolvedDependency dependency) {
                 bodyStream = Files.newInputStream(Paths.get(URI.create(uri)));
             } else {
                 HttpSender.Request.Builder request = applyAuthentication(dependency.getRepository(), httpSender.get(uri));
-                try (HttpSender.Response response = sendRequest.apply(request.build());
+                try (HttpSender.Response response = Failsafe.with(retryPolicy).get(() -> httpSender.send(request.build()));
```

Comment:
Would it make sense to additionally catch `FailsafeException` below and throw it's cause when encountered? Just to make sure that when `ExceptionUtils.sanitizeStackTrace` truncates the stacktrace, that we're left with the informative bits, and not the retry.